### ISSUE TITLE
500 on localhost and production for SMTP non-GMAIL

### ISSUE THREAD

**Issue by danieldanielecki** [dword-design/nuxt-mail#210]
Hey, I'm trying to have it working in `Nuxt3` app, but even after browsing through the issues, I can't find a workable solution.

That's what I can see on production (deployed on `Firebase`, URL: https://doyban.com/contact):

```
Request URL: https://doyban.com/mail/send
Request Method: POST
Status Code: 500 Internal Server Error
Remote Address: 199.36.158.100:443
Referrer Policy: strict-origin-when-cross-origin
```
<img width="1672" alt="Screenshot 2023-11-06 at 05 50 01" src="https://github.com/dword-design/nuxt-mail/assets/12736263/4411a310-1c23-4594-ba52-d8651c3b5c3c">

It's deployed with `nuxi build` and using Firebase Functions, tried this even with `nuxi generate`, then the error was `404`. Yes, I know the docs mentions `Does not work for static sites (via nuxt generate) because the module creates a server route.`, but I wanted to debug further.

<img width="1674" alt="Screenshot 2023-11-04 at 17 51 26" src="https://github.com/dword-design/nuxt-mail/assets/12736263/1e16f160-7778-4f70-8ddc-b34ad9a3a2db">

On localhost:

```
Request URL: http://localhost:3000/mail/send
Request Method: POST
Status Code: 500 Hostname/IP does not match certificate's altnames: Host: mail.doyban.com. is not in the cert's altnames: DNS:h6.idhosting.pl
Remote Address: 127.0.0.1:3000
Referrer Policy: strict-origin-when-cross-origin
```

Note: I checked with the domain provider where mail is hosted about using `h6.idhosting.pl`, and they recommended to use `mail.doyban.com`. I got the confirmation of port `587` being the correct as well.

<img width="1669" alt="Screenshot 2023-11-04 at 17 49 39" src="https://github.com/dword-design/nuxt-mail/assets/12736263/f56a33d0-4041-4f3b-b122-4db9d34830b5">

Here's with `h6.idhosting.pl` deployed, the same effect:

<img width="1673" alt="Screenshot 2023-11-04 at 18 18 54" src="https://github.com/dword-design/nuxt-mail/assets/12736263/0b538d6c-a636-4895-a49b-43a456e5b191">

Here's with `h6.idhosting.pl` on `localhost`, `500 Can't send mail - all recipients were rejected: 554 5.7.1 <dhcp-077-248-184-129.chello.nl[77.248.184.129]>: Client host rejected: Access denied` this time.

<img width="1671" alt="Screenshot 2023-11-04 at 18 25 42" src="https://github.com/dword-design/nuxt-mail/assets/12736263/dcdaeefd-daaf-4724-9fb0-1dd515c2a2d5">

The repo is publicly available: https://github.com/Doyban/Doyban-Website, I'd appreciate help, as it must be something really trivial.

I also tried it with `gmail` and it returns `200` on `localhost`:

<img width="1671" alt="Screenshot 2023-11-06 at 06 01 25" src="https://github.com/dword-design/nuxt-mail/assets/12736263/c20cc4a9-be00-47dc-9431-443dff0bcb72">

and it was delivered:

<img width="525" alt="Screenshot 2023-11-06 at 06 03 32" src="https://github.com/dword-design/nuxt-mail/assets/12736263/146977b5-d4df-43c1-8d9a-e6aaba79a208">

Deployed with `gmail` and it returns `200` on production:

<img width="1672" alt="Screenshot 2023-11-06 at 06 06 25" src="https://github.com/dword-design/nuxt-mail/assets/12736263/a5d9de3f-3cfa-4a82-ab94-4661b978e55f">

and it was delivered:

<img width="515" alt="Screenshot 2023-11-06 at 06 06 57" src="https://github.com/dword-design/nuxt-mail/assets/12736263/329bf3f7-7e92-4830-9027-ce7915c399f9">

**Comment by danieldanielecki** [comment]
@dword-design, is it possible to include `auth` object for non-gmail SMTP?

**Comment by danieldanielecki** [comment]
Got it working, thanks to adding `auth` for non-Gmail SMTP based on my (very) old project (https://github.com/Ditectrev/ditectrev.com-v3/blob/master/server/main.ts#L214-L225) where I used `nodemailer` itself.

Additionally, my domain provider was wrong, and indeed I had to use `h6.idhosting.pl` lol!

My final solution, works on `localhost` and in prod!! Hopefully will save some time to others :)

```
[
  'nuxt-mail',
  {
     message: {
       to: 'contact@doyban.com',
     },
     smtp: {
       host: process.env.MAIL_HOST,
       port: process.env.MAIL_PORT,
       auth: {
         user: process.env.MAIL_ACCOUNT,
         pass: process.env.MAIL_PASSWORD,
       },
    },
  },
],
```

`process.env` values are of course coming from environmental variables in `.env`.
